src/main/java/com/volasoftware/tinder/controllers/AuthenticationController.java

```diff
@@ -77,13 +78,29 @@ public ResponseEntity<?> login(
       }
   )
   @GetMapping(value = "/profile")
-  public ResponseEntity<?> getProfile(
-      @ApiParam(value = "Profile details", required = true)
-      Principal principal) {
+  public ResponseEntity<?> getProfile(Principal principal) {
 
     return ResponseHandler.generateResponse(
         AccountConstant.DETAILS,
         HttpStatus.OK,
         authenticationService.getAccountByEmail(principal.getName()));
   }
+
+  @ApiOperation(value = "Update Profile of logged Account")
+  @ApiResponses(
+      value = {
+          @ApiResponse(responseCode = "200", description = "Successfully operation!"),
+          @ApiResponse(responseCode = "400", description = "Bad request!")
+      }
+  )
+  @PutMapping(value = "/profile")
+  public ResponseEntity<?> changeProfile(
+      @ApiParam(value = "Updated account request", required = true)
+      @RequestBody AccountDto accountDto, Principal principal) {
```

Comment:
Do we need to validate this accountDto? Because I think you can supply empty values without any problem.

Reply:
The suggested change was applied.